### DISCUSSION TITLE
Welcome to BudgetML Discussions!

### DISCUSSION THREAD

**Discussion by htahir1** [ebhy/budgetml#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JulesBelveze** [comment]
Hi @htahir1 , thanks for the super handy library !

I am wondering whether or not it is possible to include some extra python file when creating the Docker container?
I am attempting to infer a custom model and thus I need a bunch of files like: checkpoint, model file, config and so on..
I couldn't find anything mentioning this in the docs.

Thanks for your help :)

**Reply by htahir1** [reply]
Hi @JulesBelveze ! Currently, it is not possible to do this: The way it works is that the container itself does not contain the model or any data - just the [environment dependencies](https://github.com/ebhy/budgetml/blob/main/server/requirements.txt) and [server code](https://github.com/ebhy/budgetml/blob/main/server/app/main.py). That said, you could always create your own container and pass it in the `launch` function with the `dockerfile_path` parameter, if you wanted custom stuff in it. You can then make a copy of the server and add whatever you like in there. Maybe thats a route you might want to explore? Let me know if you need help there.

Another way to do it is to actually download the checkpoint, model_file, config etc that you need in your Predictor's `load` method and then use it later on. You can easily put all the files you need in a google cloud bucket in the same project and download them, just like the actual `predictor_class` is downloaded in the [server](https://github.com/ebhy/budgetml/blob/5a87d21454819b15977299e46370206357a3114d/server/app/load.py#L29).
 
Yet another way would be to make modifications to the library itself and that would be more involved. The part which would change is the uploading of the `predictor_class` to the Bucket and then downloading it subsequently in the startup script of the instance: Most of the code is [here](https://github.com/ebhy/budgetml/blob/5a87d21454819b15977299e46370206357a3114d/budgetml/main.py#L112). Perhaps you could contribute to the repo with your own design -> Super happy for your contributions!

Hope any/all of this helps! Let me know if you have any further issues -> If you do please feel free to open an issue rather than a discussion as that seems to more fitting on the issue board

**Reply by JulesBelveze** [reply]
Thanks a lot for your quick answer @htahir1 !
I will look into your suggestions !

Will open an issue right away :)
